src/test/java/org/assertj/core/api/assumptions/Assumptions_assumeThat_Test.java

```diff
@@ -0,0 +1,354 @@
+            assumeThat(actual).hasPort(9090);
+          }
+
+          @Override
+          public void runPassingAssumption() {
+            assumeThat(actual).hasNoPort();
+          }
+        } },
+        { new AssumptionRunner<Future<?>>(mock(Future.class)) {
+          @Override
+          public void runFailingAssumption() {
+            assumeThat(actual).isDone();
+          }
+
+          @Override
+          public void runPassingAssumption() {
+            assumeThat(actual).isNotDone();
+          }
+        } },
+        { new AssumptionRunner<Iterable<Integer>>(asList(2, 4, 2)) {
+          @Override
+          public void runFailingAssumption() {
+            assumeThat(actual).containsOnlyOnce(2);
+          }
+
+          @Override
+          public void runPassingAssumption() {
+            assumeThat(actual).containsOnlyOnce(4);
+          }
+        } },
+        { new AssumptionRunner<Iterator<Integer>>(asList(2, 4, 2).iterator()) {
+          @Override
+          public void runFailingAssumption() {
+            assumeThat(actual).containsOnlyOnce(2);
+          }
+
+          @Override
+          public void runPassingAssumption() {
+            assumeThat(actual).containsOnlyOnce(4);
+          }
+        } },
+        { new AssumptionRunner<List<Integer>>(asList(2, 4, 2)) {
+          @Override
+          public void runFailingAssumption() {
+            assumeThat(actual).containsOnlyOnce(2);
+          }
+
+          @Override
+          public void runPassingAssumption() {
+            assumeThat(actual).containsOnlyOnce(4);
+          }
+        } },
+        { new AssumptionRunner<List<Integer>>(asList(2, 4, 2)) {
+          @Override
+          public void runFailingAssumption() {
+            assumeThat(actual).containsOnlyOnce(4).toAssert(2, "test").isNull();
+          }
+
+          @Override
+          public void runPassingAssumption() {
+            assumeThat(actual).containsOnlyOnce(4).toAssert(2, "").isEqualTo(2);
+          }
+        } },
+        { new AssumptionRunner<Map<Integer, Integer>>(Maps.newHashMap(2, 4)) {
+          @Override
+          public void runFailingAssumption() {
+            assumeThat(actual).containsKeys(4);
+          }
+
+          @Override
+          public void runPassingAssumption() {
+            assumeThat(actual).containsKeys(2);
+          }
+        } }
+    };
+  }
+
+  private static URL createUrl() {
+    try {
+      return new URL("http://example.com/pages/");
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void should_ignore_test_when_assumption_fails() {
+    assumptionRunner.runFailingAssumption();
+    assertThat(true).isFalse();
+  }
+
+  @Test
+  public void should_run_test_when_assumption_passes() {
+    assumptionRunner.runPassingAssumption();
+    ranTests++;
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    assertThat(ranTests).isEqualTo(provideAssumptionsRunners().length);
```

Comment:
add `.as("number of tests run")`, I was running a single test that succeeded but the junit report showed : `org.junit.ComparisonFailure: expected:<[1]> but was:<[0]>` which was really confusing until I realized the assertion here ;-)
also move this method just before the tests method, it will make it easier to understand the test behavior.